### PR TITLE
Add Golang.Go version 1.18

### PR DESCRIPTION
**Pull request by marcauberer**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/54491)

### CODE REVIEW

File: manifests/g/GoLang/Go/1.18.0/GoLang.Go.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- silentWithProgress
+Installers:
+- InstallerLocale: en-US
+  Architecture: x64
+  InstallerType: msi
+  Scope: machine
+  InstallerUrl: https://dl.google.com/go/go1.18.windows-amd64.msi
+  InstallerSha256: 57cba0010ce3bc8012cfa68ab0e11181ccd8493a66b109a5b86998e5d2b38ae1
+  UpgradeBehavior: install
+  ProductCode: '{E5DC7A79-C9A0-4472-8159-9F331CAB8E81}'
```
Comment (ItzLevvie) [inline manifests/g/GoLang/Go/1.18.0/GoLang.Go.installer.yaml:19]:
```suggestion
  ProductCode: '{F307E329-805A-4C79-BAEC-7FB35F3FE64B}'
```
winget-create or YamlCreate.ps1 is recommended to make new packages or new versions.

Both of these will update the ProductCode for you. ProductCode is unique for each release of the Go installer and cannot be the same.

File: manifests/g/GoLang/Go/1.18.0/GoLang.Go.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- InstallerLocale: en-US
+  Architecture: x86
+  InstallerType: msi
+  Scope: machine
+  InstallerUrl: https://dl.google.com/go/go1.18.windows-386.msi
+  InstallerSha256: 33f0d72f645d31256873225385700fa32ddb572acef8304548b869823384abd7
+  UpgradeBehavior: install
+  ProductCode: '{F03216DF-2EBC-46AA-B28C-F46B336F786C}'
```
Comment (ItzLevvie) [inline manifests/g/GoLang/Go/1.18.0/GoLang.Go.installer.yaml:27]:
```suggestion
  ProductCode: '{96064C04-3021-4953-A847-E05D1F5BABF3}'
```
winget-create or YamlCreate.ps1 is recommended to make new packages or new versions.

Both of these will update the ProductCode for you. ProductCode is unique for each release of the Go installer and cannot be the same.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

**Review by ItzLevvie — APPROVED** [review]

**Comment by marcauberer** [comment]
I assume this is a pipeline bug. Seems like the pipeline is unhealthy at the moment, as the badge above suggests ...
So I am waiting for the pipeline to re-run the check

**Pull request merged**